lib/combineNodes.js

```diff
@@ -0,0 +1,290 @@
+        var premultipliedSource = premultiplyTransformWithBuffer(transform, accessorSource, attributeId, attributeAccessor.min, attributeAccessor.max);
+        var premultipliedSourceLength = premultipliedSource.length;
+
+        var newAccessorId = createNewId(gltf, meshId, attributeId, 'accessor');
+        var newBufferId = createNewId(gltf, meshId, attributeId, 'buffer');
+        var newBufferViewId = createNewId(gltf, meshId, attributeId, 'bufferView');
+
+        var newBuffer = {
+            "type": "arraybuffer",
+            "byteLength": premultipliedSourceLength,
+            "uri": "data:,",
+            "extras": {
+                "_pipeline": {
+                    "extension": ".bin",
+                    "source": premultipliedSource,
+                    "deleteExtras": true
+                }
+            }
+        };
+        var newBufferView = {
+            "buffer": newBufferId,
+            "byteOffset": 0,
+            "byteLength": premultipliedSourceLength,
+            "extras": {
+                "_pipeline": {
+                    "deleteExtras": true
+                }
+            }
+        };
+
+        if (defined(attributeBufferView.target)) {
+            newBufferView.target = attributeBufferView.target;
+        }
+
+        var newAccessor = clone(attributeAccessor, true);
+        newAccessor.byteOffset = 0;
+        newAccessor.bufferView = newBufferViewId;
+
+        attributes[attributeId] = newAccessorId;
+        accessors[newAccessorId] = newAccessor;
+        bufferViews[newBufferViewId] = newBufferView;
+        buffers[newBufferId] = newBuffer;
+    }
+}
+
+//Returns a new buffer with the elements premultiplied by the transform matrix.
+function premultiplyTransformWithBuffer(matrix, buffer, attributeId, min, max) {
+    var bufferLength = buffer.length;
+    var resultBuffer = new Buffer(bufferLength);
+    var bufferIndex = 0;
+
+    while (bufferIndex < bufferLength) {
+        var currentVector = new Cartesian3(buffer.readFloatLE(bufferIndex), buffer.readFloatLE(bufferIndex + 4), buffer.readFloatLE(bufferIndex + 8));
+        var newVector = new Cartesian3();
+
+        if (attributeId === 'POSITION') {
+            Matrix4.multiplyByPoint(matrix, currentVector, newVector);
+        } else if (attributeId === 'NORMAL') {
+            var inverse = Matrix4.inverse(matrix, new Matrix4());
+            var inverseTranspose = Matrix4.transpose(inverse, new Matrix4());
+            Matrix4.multiplyByPointAsVector(inverseTranspose, currentVector, newVector);
+        } else {
+            Matrix4.multiplyByPointAsVector(matrix, currentVector, newVector);
+        }
+
+        resultBuffer.writeFloatLE(newVector.x, bufferIndex);
+        resultBuffer.writeFloatLE(newVector.y, bufferIndex + 4);
+        resultBuffer.writeFloatLE(newVector.z, bufferIndex + 8);
+
+        //Update the min and max values if defined.

```

Comment:
This can be a lot less code by making `min`/`max` a `Cartesian3` (convert back to an array later) and using [Cartesian3.minimumByComponent](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/Cartesian3.js#L249).